### PR TITLE
[MM-24664] Plugin_store queries squirrel refactor

### LINKED ISSUE #14424: Refactor SQL queries in store/sqlstore/plugin_store.go to use the squirrel builder
Mattermost is replacing the SQL queries of the store layer to use the [squirrel builder](https://github.com/Masterminds/squirrel). This help wanted issue is to change the usages of raw SQL queries in the file `store/sqlstore/plugin_store.go`.

The expected way to implement this is to search in the `store/sqlstore/plugin_store.go` file for all the strings containing a SQL query and rewrite them with squirrel. To do so, we must obtain the query builder using the `getQueryBuilder()` method from the store struct, and use it to build the same query that we have in the string.

You can find more information on how to use squirrel in [the official package documentation](https://godoc.org/github.com/Masterminds/squirrel).

Example Pull Request: https://github.com/mattermost/mattermost-server/pull/14393

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` in https://community.mattermost.com/

---
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-24664

### PR DESCRIPTION
**Pull request by joshuabezaleel**
#### Summary
Refactor SQL queries in plugin_store.go to use squirrel's queryBuilder.

#### Ticket Link
Fixes #14424

### CODE REVIEW

**Comment by joshuabezaleel** [comment]
Hi @mgdelacroix , I made one assumption on this.
In every query building phase, I made it into two phase where (1) the translation to SQL is made in the second phase and (2) when the `ToSql()` errors, I always use the error message of `store.sql.build_query.app_error` to return the error.

Looking forward to the review! :slightly_smiling_face:

**Comment by agnivade** [comment]
Hi @joshuabezaleel - a lot of tests seem to be failing. Once we get those passing, we can get to the review. Thanks for your contribution !

**Comment by agnivade** [comment]
Hi @joshuabezaleel - wondering if you had a chance to look at the failures ? Let us know if we can help in any way.

**Comment by joshuabezaleel** [comment]
Hi @agnivade ! So sorry that I just reviewed it now and so many errors got through. Should be okay by now. Looking forward to the review! :slightly_smiling_face:

File: store/sqlstore/plugin_store.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"fmt"
 	"net/http"
 
+	sq "github.com/Masterminds/squirrel"
```
Comment (agnivade) [inline store/sqlstore/plugin_store.go:12]: Nit: this should be in a separate import group.

Comment (joshuabezaleel) [reply store/sqlstore/plugin_store.go:12]: Fixed, thank you for the review @agnivade ! :slightly_smiling_face:

**Pull request merged**